library/src/main/camera2/com/google/android/cameraview/Camera2.java

```diff
@@ -478,15 +585,46 @@ private void startCaptureSession() {
         mSurfaceInfo.surface.setDefaultBufferSize(previewSize.getWidth(), previewSize.getHeight());
         Surface surface = new Surface(mSurfaceInfo.surface);
         try {
+            CameraCharacteristics characteristics = mCameraManager.getCameraCharacteristics(mCameraId);
+            StreamConfigurationMap map = characteristics
+                    .get(CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP);
+            mVideoSize = chooseVideoSize(map.getOutputSizes(MediaRecorder.class), mMinVideoWidth, mMinVideoHeight);
             mPreviewRequestBuilder = mCamera.createCaptureRequest(CameraDevice.TEMPLATE_PREVIEW);
             mPreviewRequestBuilder.addTarget(surface);
-            mCamera.createCaptureSession(Arrays.asList(surface, mImageReader.getSurface()),
-                    mSessionCallback, null);
+            if (mStartVideoRecording) {
+                mPreviewRequestBuilder.addTarget(mMediaRecorder.getSurface());
+                mCamera.createCaptureSession(Arrays.asList(surface, mMediaRecorder.getSurface()),
+                                             mSessionCallback, null);
+            } else {
+                mCamera.createCaptureSession(Arrays.asList(surface, mImageReader.getSurface()),
+                                             mSessionCallback, null);
+            }
         } catch (CameraAccessException e) {
             throw new RuntimeException("Failed to start camera session");
         }
     }
 
+    private android.util.Size chooseVideoSize(android.util.Size[] choices, int width, int height) {
```

Comment:
They use is a custom Size class, too. To be consistent with the rest I used the package name  to specify I wanna use the one from utils. They've done that, too.